text/0000-light-dom.md

```diff
@@ -0,0 +1,229 @@
+
+- **Scoped Styles**
+
+  Shadow DOM styles are scoped to the enclosing shadow tree. Styles don't leak out to the rest of the page and the page styles don't leak into this component. In Native shadow, it is enforced by the browser whereas in Synthetic shadow it is enforced by adding few attributes to the elements and scoping all CSS rules by those attributes.
+
+  Styles scoping in Light DOM will be done differently. The styles will be scoped not just to the component they belong, but also to its children. These styles won't leak out to the parents (or their children) though.
+
+  E.g.
+
+  ```html
+  <x-a>
+    <style>
+      p {
+        color: blue;
+      }
+    </style>
+
+    <p></p> <!-- will be blue -->
+    <p class="red"></p> <!-- will still be blue. red class from child won't leak out here -->
+
+    <x-b>
+      <style>
+        p.red {
+          color: red;
+        }
+      </style>
+
+      <p></p> <!-- will be blue. Style from x-a will be applied -->
+      <p class="red"></p> <!-- will be red -->
+      <p class="red"></p> <!-- This paragraph is coming from A and slotted into B. Will also be red. -->
+    </x-b>
+  </x-a>
+  ```
+
+  Opting out of this scoping is not supported. There's no way for a component author to say a CSS rule shouldn't be scoped to that specific component (and its children). If global scoping is desired, a global stylesheet can be injected manually.
+
+- **`this.template`**
+
+  In `LightningElement`, `this.template` returns the shadow-root. It will return `null` in `MacroElement`.
+
+- **Querying**
+
+  `MacroElement` (and `LightningElement`) forward several DOM querying/manipulation methods like `querySelector`, `dispatchEvent` etc. to the host element. Full list of methods forwarded [here](https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine-core/src/framework/base-lightning-element.ts#L242). 
+  
+  This will allow component author to just use `this.querySelector` instead of using `this.template.querySelector` in case of `LightningElement`.
+  However, there's no way to get a direct reference to the host element, like `this.template.host` that's available in `LightningElement`.
+
+  E.g.
+  ```js
+  /* <template>
+   *   <x-child></x-child>
+   * </template>
+   */
+  export default class ParentElement extends MacroElement {
+    renderedCallback() {
+      this.querySelector('x-child').addEventListener('custom', (e) => {
+          console.log('Called with ' + JSON.stringify(e.detail));
+      });
+    }
+  }
+
+  /**
+   * <template>
+   *   <button>Click Me</button>
+   * </template>
+   */
+  export default class ChildElement extends MacroElement {
+      renderedCallback() {
+          this.querySelector('button').addEventListener('click', this.handleClick);
+      }
+
+      handleClick() {
+        this.dispatchEvent(new CustomEvent('custom', { detail: { macro: true } }));
```

Comment:
given how the event listener is added, what's the expected `this` within the event handler?

Reply:
In this example, the `this` value is the `button` element. Event listeners added via the template are the only ones binding the `this` value to the class.